components/backup-stream/src/subscription_track.rs

```diff
@@ -163,59 +243,54 @@ impl SubscriptionTracer {
         let rs = regions.into_iter().collect::<HashSet<_>>();
         self.0
             .iter_mut()
-            .filter(|s| {
-                let contains = rs.contains(s.key());
-                if !contains {
-                    crate::metrics::LOST_LEADER_REGION.inc();
+            // Don't advance the checkpoint ts of pending region.
+            .filter_map(|mut s| {
+                let region_id = *s.key();
+                match s.value_mut() {
+                SubscribeState::Running(sub) => {
+                    let contains = rs.contains(&region_id);
+                    if !contains {
+                        crate::metrics::LOST_LEADER_REGION.inc();
+                    }
+                    contains.then(|| ResolveResult::resolve(sub, min_ts))
                 }
-                contains
+                SubscribeState::Pending(r) => {warn!("pending region, skip resolving"; utils::slog_region(r)); None},
```

Comment:
how about use metrics instead of log here?

Reply:
I guess this won't be so verbose and provide little information without context. 🤔

Reply:
ok, looks like it's not easy to trigger Pending region resolve